forc-pkg/src/lock.rs

```diff
@@ -191,7 +197,14 @@ impl Lock {
                 .as_ref()
                 .into_iter()
                 .flatten()
-                .map(|contract_dep| (contract_dep, DepKind::Contract));
+                .map(|contract_dep| {
+                    (
+                        contract_dep,
+                        DepKind::Contract {
+                            salt: fuel_tx::Salt::zeroed(),
```

Comment:
I'm a bit weary of using a fake `Salt` here, even if it's only temporary. We should either:

- Comment very clearly explaining that this value is overridden later on in the function or
- Make a dedicated enum type just for distinguishing between the dep kinds here. E.g.
  ```rust
  enum UnparsedDepKind { Library, Contract }
  ```
  then this could be:
  ```rust
  .map(|contract_dep| (contract_dep, UnparsedDepKind::Contract))
  ```

My preference would be for the latter, as it ensures a bit better type safety, and avoids the possibility of accidentally using the invalid salt in future refactors.